internal/certdb/validation_test.go

```diff
@@ -0,0 +1,160 @@
+Sgi7T+lz+c1Tw+XFgkqryUwMeG2wxt8=
+-----END CERTIFICATE REQUEST-----
+`
+
+var ValidCert2 string = `-----BEGIN CERTIFICATE-----
+MIIDrDCCApSgAwIBAgIURKr+jf7hj60SyAryIeN++9wDdtkwDQYJKoZIhvcNAQEL
+BQAwOTELMAkGA1UEBhMCVVMxKjAoBgNVBAMMIXNlbGYtc2lnbmVkLWNlcnRpZmlj
+YXRlcy1vcGVyYXRvcjAeFw0yNDAzMjcxMjQ4MDRaFw0yNTAzMjcxMjQ4MDRaMEcx
+FjAUBgNVBAMMDTEwLjE1Mi4xODMuNTMxLTArBgNVBC0MJDM5YWNlMTk1LWRjNWEt
+NDMyYi04MDkwLWFmZTZhYjRiNDljZjCCASIwDQYJKoZIhvcNAQEBBQADggEPADCC
+AQoCggEBAIzOVs/h0bQ73kc3g5BAzOKK5yGnn3vHfJ5hYqVGrfaglN1Pe/fBUeOt
+Ax4wixsRccZqEYrcx7WkeHsOR9N+h5RgdtMmrPwjA4ElCOvQ8gLMj8BCajlgRjtS
+y+dKl3Zef4zpDtr6APY+VLA9QrVkpJcoWzYVjKWKo9jm1l3GmI/Rsf08GIcPMbCf
+0EVsgq7441cx/PuH63w4QVTsssDe7KviN1sAm8gQXDfLjsmntG/6uflCVIqoKXjg
+If+GO6iN2GqzI+HJu3cJoglY3nVMIeuc1ch7UFHy3fJxVipfe++ZjQXNconLyg1N
+2QeYWdTGbaWzAcGgNcjEhK7P34eSml8CAwEAAaOBnTCBmjAhBgNVHSMEGjAYgBYE
+FN/vgl9cAapV7hH9lEyM7qYS958aMB0GA1UdDgQWBBRJJDZkHr64VqTC24DPQVld
+Ba3iPDAMBgNVHRMBAf8EAjAAMEgGA1UdEQRBMD+CN3ZhdWx0LWs4cy0wLnZhdWx0
+LWs4cy1lbmRwb2ludHMudmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWyHBAqYtzUwDQYJ
+KoZIhvcNAQELBQADggEBAEH9NTwDiSsoQt/QXkWPMBrB830K0dlwKl5WBNgVxFP+
+hSfQ86xN77jNSp2VxOksgzF9J9u/ubAXvSFsou4xdP8MevBXoFJXeqMERq5RW3gc
+WyhXkzguv3dwH+n43GJFP6MQ+n9W/nPZCUQ0Iy7ueAvj0HFhGyZzAE2wxNFZdvCs
+gCX3nqYpp70oZIFDrhmYwE5ij5KXlHD4/1IOfNUKCDmQDgGPLI1tVtwQLjeRq7Hg
+XVelpl/LXTQawmJyvDaVT/Q9P+WqoDiMjrqF6Sy7DzNeeccWVqvqX5TVS6Ky56iS
+Mvo/+PAJHkBciR5Xn+Wg2a+7vrZvT6CBoRSOTozlLSM=
+-----END CERTIFICATE-----`
+
+func TestCSRValidationSuccess(t *testing.T) {
+	if err := certdb.ValidateCertificateRequest(ValidCSR1); err != nil {
+		t.Fatalf("Couldn't verify valid CSR: %s", err)
+	}
+	if err := certdb.ValidateCertificateRequest(ValidCSR2); err != nil {
+		t.Fatalf("Couldn't verify valid CSR: %s", err)
+	}
+	if err := certdb.ValidateCertificateRequest(ValidCSR3); err != nil {
+		t.Fatalf("Couldn't verify valid CSR: %s", err)
+	}
+}
+
+func TestCSRValidationFail(t *testing.T) {
+	var wrongString string = "this is a real csr!!!"
+	err := certdb.ValidateCertificateRequest(wrongString)
+	if err.Error() != "PEM Certificate Request string not found or malformed" {
+		t.Fatalf("Expected error not found:\nReceived: %s", err)
+	}
+	var ValidCSRWithoutWhitespace = strings.ReplaceAll(ValidCSR1, "\n", "")
+	err = certdb.ValidateCertificateRequest(ValidCSRWithoutWhitespace)
+	if err.Error() != "PEM Certificate Request string not found or malformed" {
+		t.Fatalf("Expected error not found:\nReceived: %s", err)
+	}
+	var wrongPemType string = strings.ReplaceAll(ValidCSR1, "CERTIFICATE REQUEST", "SOME RANDOM PEM TYPE")
+	err = certdb.ValidateCertificateRequest(wrongPemType)
+	if err.Error() != "given PEM string not a certificate request" {
+		t.Fatalf("Expected error not found:\nReceived: %s", err)
+	}
+	var InvalidCSR = strings.ReplaceAll(ValidCSR1, "/", "p")
+	err = certdb.ValidateCertificateRequest(InvalidCSR)
+	if err == nil {
+		t.Fatalf("Expected CSR to fail validation")
+	}
+}
+
+// Fuzz test
```

Comment:
There are some comments here that I am not sure are needed, are they required?

Reply:
I was planning on doing some extra stuff, but they're overkill especially since they're just gonna test golang stdlib instead of the validations. removed comments.